src/Core/Services/PasswordGenerationService.cs

```diff
@@ -51,52 +51,7 @@ public async Task<string> GeneratePasswordAsync(PasswordGenerationOptions option
             }
 
             // Sanitize
-            if(options.Uppercase.GetValueOrDefault() && options.MinUppercase.GetValueOrDefault() <= 0)
-            {
-                options.MinUppercase = 1;
-            }
-            else if(!options.Uppercase.GetValueOrDefault())
-            {
-                options.MinUppercase = 0;
-            }
-
-            if(options.Lowercase.GetValueOrDefault() && options.MinLowercase.GetValueOrDefault() <= 0)
-            {
-                options.MinLowercase = 1;
-            }
-            else if(!options.Lowercase.GetValueOrDefault())
-            {
-                options.MinLowercase = 0;
-            }
-
-            if(options.Number.GetValueOrDefault() && options.MinNumber.GetValueOrDefault() <= 0)
-            {
-                options.MinNumber = 1;
-            }
-            else if(!options.Number.GetValueOrDefault())
-            {
-                options.MinNumber = 0;
-            }
-
-            if(options.Special.GetValueOrDefault() && options.MinSpecial.GetValueOrDefault() <= 0)
-            {
-                options.MinSpecial = 1;
-            }
-            else if(!options.Special.GetValueOrDefault())
-            {
-                options.MinSpecial = 0;
-            }
-
-            if(options.Length.GetValueOrDefault() < 1)
-            {
-                options.Length = 10;
-            }
-            var minLength = options.MinSpecial.GetValueOrDefault() + options.MinLowercase.GetValueOrDefault() +
-                options.MinNumber.GetValueOrDefault() + options.MinUppercase.GetValueOrDefault();
-            if(options.Length < minLength)
-            {
-                options.Length = minLength;
-            }
+            SanitizePasswordLength(options, true);
```

Comment:
This looks like vince's stuff from a previous commit. What happened here?

Reply:
Good question - I've been rebasing but maybe something got screwy - let me try again.

Reply:
Found the problem locally but now it's complaining about the branches diverging and won't let me push until I pull - which will re-introduce the problem.  Not sure how to fix this properly so I may just start fresh.